src/crafting_gui.cpp

```diff
@@ -48,6 +51,40 @@ void load_recipe_category( JsonObject &jsobj )
     }
 }
 
+std::string get_cat_name( std::string prefixed_name )
+{
+    if( prefixed_name.find("CC_") == 0 ) {
+        return prefixed_name.substr( 3, prefixed_name.size() - 3 );
+    }
+
+    debugmsg( ( "could not get screen name of: " + prefixed_name ).c_str() );
+    return prefixed_name;
+}
+
+std::string get_subcat_name( const std::string &cat, std::string prefixed_name )
+{
+    std::string prefix = "CSC_" + get_cat_name( cat ) + "_";
+
+    if( prefixed_name.find( prefix ) == 0 ) {
+        return prefixed_name.substr( prefix.size(), prefixed_name.size() - prefix.size() );
+    } else if( prefixed_name.find("CSC_ALL") == 0 ) {
+        return "ALL";
+    }
+
+    debugmsg( ( "could not get screen name of: " + prefixed_name ).c_str() );
+    return prefixed_name;
+}
+
+void translate_all() {
```

Comment:
Ooh that's new :)

Ah, good to know that the strings in jason are automatically translated. And I can retrieve it using ```_()```. But that doesn't make the map redundant, it's still needed to store the normalized form of the category identifier. Code still needs to be changed thought.

e.g. : A category idenftier is stored in json like ```CC_WEAPON```, and automatically translated. In the functions I added, the string is normalized into ```WEAPON``` and then stored in the translation map (I should use ```_()``` on the raw string, not the normalized one). We need the map to go from ```CC_WEAPON``` to ```WEAPON``` later, because the current tab is denoted by a string of the ```CC_``` form. (maybe I should rename the 'translations' map to 'normalized_names').